File: Lab2/FileReverse.java
//-----------------------------------------------------------------------------
// Charishma Thota
// cthota
// FileReverse.java
// Reads a text file and reverses every word in that file and inputs it onto another text file
//-----------------------------------------------------------------------------
import java.io.*;
import java.util.Scanner;
import java.util.*;

class FileReverse{
	public static void main(String[] args) throws IOException{

	

	// check number of command line arguments is at least 2
	if(args.length < 2){	
	System.out.println("Usage: java –jar FileTokens.jar <input file> <output file>");
        System.exit(1);
	}

	// open files
	Scanner in = new Scanner(new File(args[0]));
	PrintWriter out = new PrintWriter(new FileWriter(args[1]));

	// read lines from in, extract and print tokens from each line
	while( in.hasNext() ){
	String word = in.next();
	out.println(stringReverse(word));
	}
	// close files
	in.close();
	out.close();
}

	public static String stringReverse(String s){
	int letterCount = s.length() - 1; // counts the amount of letters in a word
	// create two arrays that will store current word and reversed word
	char[] wordReversed = new char[ letterCount + 1]; 
	char[] currentWord = s.toCharArray(); 
	for( int i = letterCount; i>=0; i--){	// loop around the char array to reverse it
		wordReversed[letterCount - i] = currentWord[i];
		}
	// convert char array to String to match return type
	String revWord = new String(wordReversed);
	return revWord;
	}
}

File: Lab3/fileReverse.c
/*
* Charishma Thota
* cthota
*  fileReverse.c
*  Reads input file and prints each word on a separate line of
*  the output file.
*/
# include <stdio.h>
# include <stdlib.h>
# include <string.h>

void stringReverse(char* s);

int main(int argc, char* argv[]){
   FILE* in;  /* file handle for input */  
   FILE* out; /* file handle for output */
   char word[256]; /* char array to store words from input file */

   /* check command line for correct number of arguments */
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(EXIT_FAILURE);
   }

   /* open input file for reading */
   in = fopen(argv[1], "r");
   if( in==NULL ){
      printf("Unable to read from file %s\n", argv[1]);
      exit(EXIT_FAILURE);
   }

   /* open output file for writing */
   out = fopen(argv[2], "w");
   if( out==NULL ){
      printf("Unable to write to file %s\n", argv[2]);
      exit(EXIT_FAILURE);
   }

   /* read words from input file, print on separate lines to output file*/
   while( fscanf(in, " %s", word) != EOF) {
       stringReverse(word);
      fprintf(out, "%s\n", word);
   }

   /* close input and output files */
   fclose(in);
   fclose(out);

   return(EXIT_SUCCESS);
}
void stringReverse(char* s) {
    size_t stringLen= strlen(s)-1; // get the length of string
    size_t j = stringLen;
    char temp; // to swap the letters

    for(size_t i = 0; i < stringLen; i++){
        // swap here
        temp = s[j]; // use an array to get the letter
        s[j] = s[i];
        s[i] = temp;
        j--;

        if(j == (stringLen/2)){ // you only have to go through half of the characters in the word in order to reverse
            break; // so break out of the for loop after going through half
        }
        
    }
  
}
